cf_xarray/accessor.py

```diff
@@ -2254,23 +2268,33 @@ def decode_vertical_coords(self, prefix="z"):
                     terms["depth_c"] * terms["s"]
                     + (terms["depth"] - terms["depth_c"]) * terms["C"]
                 )
+
                 # z(n,k,j,i) = S(k,j,i) + eta(n,j,i) * (1 + S(k,j,i) / depth(j,i))
-                ds.coords[zname] = S + terms["eta"] * (1 + S / terms["depth"])
+                ztemp = terms["eta"] * (1 + S / terms["depth"]) + S
 
             elif stdname == "ocean_s_coordinate_g2":
                 # make sure all necessary terms are present in terms
                 # (depth_c * s(k) + depth(j,i) * C(k)) / (depth_c + depth(j,i))
                 S = (terms["depth_c"] * terms["s"] + terms["depth"] * terms["C"]) / (
                     terms["depth_c"] + terms["depth"]
                 )
+
                 # z(n,k,j,i) = eta(n,j,i) + (eta(n,j,i) + depth(j,i)) * S(k,j,i)
-                ds.coords[zname] = terms["eta"] + (terms["eta"] + terms["depth"]) * S
+                ztemp = terms["eta"] + (terms["eta"] + terms["depth"]) * S
+
+            elif stdname == "ocean_sigma_coordinate":
+                # z(n,k,j,i) = eta(n,j,i) + sigma(k)*(depth(j,i)+eta(n,j,i))
+                ztemp = terms["eta"] + terms["sigma"] * (terms["depth"] + terms["eta"])
 
             else:
                 raise NotImplementedError(
                     f"Coordinate function for {stdname!r} not implemented yet. Contributions welcome!"
                 )
 
+            # add Axis attribute
+            ztemp.attrs["axis"] = "Z"
```

Comment:
In the CIOFS ROMS output I am looking at (NOAA model), `s_rho` has `positive : up` in its attributes so it is identified as being a vertical coordinate. So that seems to make it more complicated for being able to label the z coordinate with `positive: up`. Or, maybe that label is found on the "s" coordinate in the formula, it can be removed from there and added to the resulting z coordinate?

Reply:
Ah that complicates things. Lets just skip adding attributes.